src/OrbitGl/TrackContainer.cpp

```diff
@@ -370,11 +369,66 @@ void TrackContainer::SetIteratorOverlayData(
   RequestUpdate(RequestUpdateScope::kDraw);
 }
 
+void TrackContainer::DrawThreadDependency(PrimitiveAssembler& primitive_assembler,
+                                          PickingMode picking_mode) {
+  std::optional<orbit_client_data::ThreadStateSliceInfo> thread_state_slice =
+      app_->selected_thread_state_slice();
+  if (!thread_state_slice.has_value() || picking_mode != PickingMode::kNone) {
+    return;
+  }
+
+  orbit_client_data::ThreadID start_arrow_thread = thread_state_slice->wakeup_tid();
+  orbit_client_data::ThreadID end_arrow_thread = thread_state_slice->tid();
+
+  std::optional<ThreadTrack*> temp_start_arrow_track =
+      track_manager_->GetThreadTrack(start_arrow_thread);
+  std::optional<ThreadTrack*> temp_end_arrow_track =
+      track_manager_->GetThreadTrack(end_arrow_thread);
+
+  if (!temp_end_arrow_track.has_value() || !temp_start_arrow_track.has_value()) {
+    return;
+  }
+
+  ThreadTrack* start_arrow_track = temp_start_arrow_track.value();
+  ThreadTrack* end_arrow_track = temp_end_arrow_track.value();
+
+  const std::vector<Track*>& threads = track_manager_->GetVisibleTracks();
+
+  if (std::find(threads.begin(), threads.end(), start_arrow_track) == threads.end() ||
+      std::find(threads.begin(), threads.end(), end_arrow_track) == threads.end()) {
+    return;
+  }
+
+  const Color kArrowColor(255, 255, 255, 255);
+  const float kOffsetToGetCenterOfEndingThreadStateBar =
+      end_arrow_track->GetThreadStateBarHeight() / 2;
+  const float kOffsetToGetCenterOfStartingThreadStateBar =
+      start_arrow_track->GetThreadStateBarHeight() / 2;
+  float x = timeline_info_->GetWorldFromTick(thread_state_slice->begin_timestamp_ns());
+  float y_start =
+      start_arrow_track->GetThreadStateBarPos()[1] + kOffsetToGetCenterOfEndingThreadStateBar;
+  float y_end =
+      end_arrow_track->GetThreadStateBarPos()[1] + kOffsetToGetCenterOfStartingThreadStateBar;
+  float whole_height = std::abs(y_end - y_start);
+  float body_height = whole_height - layout_->GetThreadDependencyArrowHeadHeight();
+  Vec2 arrow_start_pos{x, y_start};
+  Vec2 arrow_body_size{layout_->GetThreadDependencyArrowBodyWidth(), body_height};
+  Vec2 arrow_head_size{layout_->GetThreadDependencyArrowHeadWidth(),
```

Comment:
In general I'm not fan to create this variables when the naming is easy to understand. Please consider including this in AddVerticalArrow directly.

`arrow_direction` is a different case, since it includes some logic.